clients/src/main/java/org/apache/kafka/clients/InFlightRequests.java

```diff
@@ -55,13 +55,20 @@ public void add(ClientRequest request) {
     }
 
     /**
-     * Get the oldest request (the one that that will be completed next) for the given node
+     * Get the oldest request (the one that that will be completed next) for the given node and remove it from the queue
      */
     public ClientRequest completeNext(String node) {
         return requestQueue(node).pollLast();
     }
 
     /**
+     * Get the oldest request (the one that that will be completed next) for the given node
+     */
+    public ClientRequest earliestSent(String node) {
```

Comment:
Why is this called `earliestSent`? We have `lastSent` and `completeLastSent`, so `next` would go with `completeNext`?